src/device-state.ts

```diff
@@ -42,52 +44,17 @@ import * as apiKeys from './lib/api-keys';
 
 const disallowedHostConfigPatchFields = ['local_ip', 'local_port'];
 
-function validateLocalState(state: any): asserts state is TargetState['local'] {
-	if (state.name != null) {
-		if (!validation.isValidShortText(state.name)) {
-			throw new Error('Invalid device name');
-		}
-	}
-	if (state.apps == null || !validation.isValidAppsObject(state.apps)) {
-		throw new Error('Invalid apps');
-	}
-	if (state.config == null || !validation.isValidEnv(state.config)) {
-		throw new Error('Invalid device configuration');
-	}
-}
+function parseTargetState(state: unknown): TargetState {
+	const res = TargetState.decode(state);
 
-function validateDependentState(
-	state: any,
-): asserts state is TargetState['dependent'] {
-	if (
-		state.apps != null &&
-		!validation.isValidDependentAppsObject(state.apps)
-	) {
-		throw new Error('Invalid dependent apps');
+	if (isRight(res)) {
+		return res.right;
 	}
-	if (
-		state.devices != null &&
-		!validation.isValidDependentDevicesObject(state.devices)
-	) {
-		throw new Error('Invalid dependent devices');
-	}
-}
 
-function validateState(state: any): asserts state is TargetState {
-	if (!_.isObject(state)) {
-		throw new Error('State must be an object');
-	}
-	// these any typings seem unnecessary but the `isObject`
-	// call above tells typescript that state is of type
-	// `object` - which apparently does not allow any fields
-	// to be accessed
-	if (!_.isObject((state as any).local)) {
-		throw new Error('Local state must be an object');
-	}
-	validateLocalState((state as any).local);
-	if ((state as any).dependent != null) {
-		return validateDependentState((state as any).dependent);
-	}
+	const errors = [
+		'Invalid target state. Please ensure you are using the most up-to-date version of Balena CLI.',
```

Comment:
We may need to make another PR when we know the specific balenaCLI version that moves to v3 target state

Reply:
I might remove this, the PR also converts between v2 and v3 target state compatibility in order to avoid breaking the supervisor API so it really it should not depend on the CLI version

Reply:
This message will appear if target state from the cloud is wrong for some reason. Should we just replace it with `Invalid target state`?